# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363125214 25200
#      Tue Mar 12 14:53:34 2013 -0700
# Node ID d07aeecb2d229e07fbe526325ef1b42a82ce44d8
# Parent  fe3b9a51e625a4c50c77a3c3a11c5dc243558c12
build: Add more #ifdefs to build GUI when Qscintilla is not present.

* libgui/src/m-editor/file-editor-interface.h,
libgui/src/m-editor/file-editor-tab.h, libgui/src/m-editor/file-editor.h,
libgui/src/m-editor/find-dialog.h, libgui/src/m-editor/lexer-octave-gui.h:
Surround entire file with #ifdef HAVE_QSCINTILLA so that generated moc-*.cc
file will not be present if Qscintilla is not installed.

* libgui/src/main-window.cc: Add #ifdef HAVE_QSCINTILLA around instances
of file_editor.

diff --git a/libgui/src/m-editor/file-editor-interface.h b/libgui/src/m-editor/file-editor-interface.h
--- a/libgui/src/m-editor/file-editor-interface.h
+++ b/libgui/src/m-editor/file-editor-interface.h
@@ -18,16 +18,18 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORINTERFACE_H
 #define FILEEDITORINTERFACE_H
 
+#ifdef HAVE_QSCINTILLA
+
 #include <QMenu>
 #include <QToolBar>
 #include "octave-dock-widget.h"
 
 class file_editor_interface : public octave_dock_widget
 {
   Q_OBJECT
 
@@ -56,9 +58,11 @@ public slots:
 //signals:
 
 //protected:
 
 //protected slots:
 
 };
 
-#endif // FILEEDITORINTERFACE_H
+#endif  // HAVE_QSCINTILLA
+
+#endif  // FILEEDITORINTERFACE_H
diff --git a/libgui/src/m-editor/file-editor-tab.h b/libgui/src/m-editor/file-editor-tab.h
--- a/libgui/src/m-editor/file-editor-tab.h
+++ b/libgui/src/m-editor/file-editor-tab.h
@@ -18,16 +18,18 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORTAB_H
 #define FILEEDITORTAB_H
 
+#ifdef HAVE_QSCINTILLA
+
 #include <Qsci/qsciscintilla.h>
 #include <QWidget>
 #include <QCloseEvent>
 #include <QFileSystemWatcher>
 
 #include "find-dialog.h"
 
 class file_editor;
@@ -151,9 +153,11 @@ private:
 
   QFileSystemWatcher    _file_system_watcher;
 
   find_dialog *         _find_dialog;
   bool                  _find_dialog_is_visible;
   QRect                 _find_dialog_geometry;
 };
 
-#endif // FILEEDITORTAB_H
+#endif  // HAVE_QSCINTILLA
+
+#endif  // FILEEDITORTAB_H
diff --git a/libgui/src/m-editor/file-editor.h b/libgui/src/m-editor/file-editor.h
--- a/libgui/src/m-editor/file-editor.h
+++ b/libgui/src/m-editor/file-editor.h
@@ -18,16 +18,18 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORMDISUBWINDOW_H
 #define FILEEDITORMDISUBWINDOW_H
 
+#ifdef HAVE_QSCINTILLA
+
 #include <QToolBar>
 #include <QAction>
 #include <QMenuBar>
 #include <QStatusBar>
 #include <QCloseEvent>
 #include <QTabWidget>
 
 #include "file-editor-interface.h"
@@ -149,9 +151,11 @@ private:
 
   enum { MaxMRUFiles = 10 };
   QMenu *_mru_file_menu;
   QAction *_mru_file_actions[MaxMRUFiles];
   QStringList _mru_files;
 
 };
 
-#endif // FILEEDITORMDISUBWINDOW_H
+#endif  // HAVE_QSCINTILLA
+
+#endif  // FILEEDITORMDISUBWINDOW_H
diff --git a/libgui/src/m-editor/find-dialog.h b/libgui/src/m-editor/find-dialog.h
--- a/libgui/src/m-editor/find-dialog.h
+++ b/libgui/src/m-editor/find-dialog.h
@@ -37,16 +37,18 @@
 ** Nokia at qt-info@nokia.com.
 ** $QT_END_LICENSE$
 **
 ****************************************************************************/
 
 #ifndef FIND_DIALOG_H
 #define FIND_DIALOG_H
 
+#ifdef HAVE_QSCINTILLA
+
 #include <QDialog>
 #include <Qsci/qsciscintilla.h>
 
 class QCheckBox;
 class QDialogButtonBox;
 class QGroupBox;
 class QLabel;
 class QLineEdit;
@@ -81,10 +83,12 @@ private:
   QPushButton       *_replace_button;
   QPushButton       *_replace_all_button;
   QPushButton       *_more_button;
   QWidget           *_extension;
   QsciScintilla     *_edit_area;
   bool               _find_result_available;
 };
 
-#endif // FIND_DIALOG_H
+#endif  // HAVE_QSCINTILLA
 
+#endif  // FIND_DIALOG_H
+
diff --git a/libgui/src/m-editor/lexer-octave-gui.h b/libgui/src/m-editor/lexer-octave-gui.h
--- a/libgui/src/m-editor/lexer-octave-gui.h
+++ b/libgui/src/m-editor/lexer-octave-gui.h
@@ -18,16 +18,18 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef LEXEROCTAVE_H
 #define LEXEROCTAVE_H
 
+#ifdef HAVE_QSCINTILLA
+
 #include "resource-manager.h"
 #include <QObject>
 #include <Qsci/qsciglobal.h>
 #include <Qsci/qscilexer.h>
 #include <Qsci/qsciapis.h>
 
 class lexer_octave_gui : public QsciLexer
 {
@@ -58,9 +60,11 @@ class lexer_octave_gui : public QsciLexe
   QString description (int style) const;
 
 private:
   lexer_octave_gui (const lexer_octave_gui &);
   lexer_octave_gui &operator= (const lexer_octave_gui &);
   QsciAPIs *lexer_api;
 };
 
-#endif
+#endif  // HAVE_QSCINTILLA
+
+#endif  // LEXEROCTAVE_H
diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -32,17 +32,19 @@ along with Octave; see the file COPYING.
 #include <QSettings>
 #include <QStyle>
 #include <QToolBar>
 #include <QDesktopServices>
 #include <QFileDialog>
 #include <QMessageBox>
 #include <QIcon>
 
+#ifdef HAVE_QSCINTILLA
 #include "file-editor.h"
+#endif
 #include "main-window.h"
 #include "octave-link.h"
 #include "settings-dialog.h"
 
 #include "builtins.h"
 #include "defaults.h"
 #include "load-save.h"
 #include "toplev.h"
@@ -716,17 +718,19 @@ main_window::construct ()
   QAction *new_gui_action = new_menu->addAction (tr ("GUI"));
   new_gui_action->setEnabled (false); // TODO: Make this work.
 
   QAction *open_action
     = file_menu->addAction (QIcon(":/actions/icons/fileopen.png"), tr ("Open..."));
   open_action->setShortcut (QKeySequence::Open);
   open_action->setShortcutContext (Qt::ApplicationShortcut);
 
+#ifdef HAVE_QSCINTILLA
   file_menu->addMenu(_file_editor->get_mru_menu ());
+#endif
 
   QAction *close_command_window_action
     = file_menu->addAction (tr ("Close Command Window"));
   close_command_window_action->setShortcut (QKeySequence::Close);
   close_command_window_action->setEnabled (false); // TODO: Make this work.
 
   file_menu->addSeparator (); /////
 
